crates/bevy_render/src/mesh/shape/regular_polygon.rs

```diff
@@ -0,0 +1,81 @@
+use crate::mesh::{Indices, Mesh};
+use wgpu::PrimitiveTopology;
+
+/// A regular polygon on the xy plane
+#[derive(Debug, Copy, Clone)]
+pub struct RegularPolygon {
+    /// Inscribed radius on the xy plane.
+    pub radius: f32,
+    /// Number of sides.
+    pub sides: usize,
+}
+impl Default for RegularPolygon {
+    fn default() -> Self {
+        Self {
+            radius: 0.5,
+            sides: 6,
+        }
+    }
+}
+
+impl From<RegularPolygon> for Mesh {
+    fn from(polygon: RegularPolygon) -> Self {
+        let RegularPolygon { radius, sides } = polygon;
+
+        let mut positions = Vec::with_capacity(sides);
+        let mut normals = Vec::with_capacity(sides);
+        let mut uvs = Vec::with_capacity(sides);
+
+        for i in 0..sides {
+            let a = std::f32::consts::FRAC_PI_2 - i as f32 * std::f32::consts::TAU / (sides as f32);
+
+            positions.push([a.cos() * radius, a.sin() * radius, 0.0]);
+            normals.push([0.0, 0.0, 1.0]);
+            uvs.push([(a.cos() + 1.0) / 2.0, 1.0 - (a.sin() + 1.0) / 2.0]);
```

Comment:
```suggestion
            let (s, c) = a.sin_cos();
            positions.push([c * radius, s * radius, 0.0]);
            normals.push([0.0, 0.0, 1.0]);
            uvs.push([0.5 * (c + 1.0), 1.0 - 0.5 * (s + 1.0)]);
```
Calculating sin and cos at the same time is faster. Multiplying is faster than dividing. It would probably only make a difference if you're making _many_ polygons but maybe someone is and I see no harm.

Reply:
Oh, nice. I benched this (outside of bevy, just building the vertexes and indices outside of `Mesh`) and it's good for a 3-14% speedup for 3-128 sides.